### PR TITLE
Remove UseToolVersion VS2017 from MsBuild task

### PR DESCRIPTION
**Pull request by owen-roberts**
This causes the task to fail if VS 2017 isn't installed.

### CODE REVIEW

**Comment by owen-roberts** [comment]
Added https://github.com/stormid/cake-mix/pull/18/commits/810aa33564158065328163dc26321a37bb075428 which makes the MsBuild Tool Version configurable via a script argument. 

Usage: 
` ./build.ps1 -ScriptArgs '-MsBuildToolVersion="VS2019"'`

**Comment by owen-roberts** [comment]
Add https://github.com/stormid/cake-mix/pull/18/commits/3f6347137f884c6dc44f0cbf997250afb3560a86 which moves the `MSBuildToolVersion` property on `Configuration` into a partial class within the Build-MsBuild task. 

Can now set a default value in the `build.cake` if no script argument is provided. If an argument is provided it will take priority.

File: .cake/Publish-MsBuild-Folder.cake [hunk 1]
```diff
@@ -14,7 +16,7 @@ Task("Publish:MsBuild")
         MSBuild(webProject.ProjectFilePath, c => c
             .SetConfiguration(config.Solution.BuildConfiguration)
             .SetVerbosity(Verbosity.Quiet)
-            .UseToolVersion(MSBuildToolVersion.VS2017)
+            .UseToolVersion(config.MsBuildToolVersion)
```
Comment (Philo) [inline .cake/Publish-MsBuild-Folder.cake]: Is `MsBUildToolVersion` a property added into the base `Configuration` class? - this is introducing a property into configuration that is only used by specific tasks.  The `TaskParameters` dictionary should be used to set and get task specific parameters, where the task itself is aware of a task parameter with a given key.  See the Npm-RunScript or CopyArtifacts as examples.

Comment (owen-roberts) [reply .cake/Publish-MsBuild-Folder.cake]: The `MSBuildToolVersion` is a property on a partial `Configuration` class defined in the Build-MsBuild task.

As discussed I'm going to change this to use the `TaskParameters` dictionary instead.

Comment (owen-roberts) [reply .cake/Publish-MsBuild-Folder.cake]: Fixed with https://github.com/stormid/cake-mix/pull/18/commits/bd5a5791cf7fa1b475dfa76eeb6a62bc131e89c6

**Pull request merged**